converters/provider-aws/dynamodb/dynamodb.go

```diff
@@ -47,35 +47,40 @@ func TableResource(mg resource.Managed) ([]resource.Managed, error) {
 			target.Spec.ForProvider.RangeKey = t.AttributeName
 		}
 	}
-	for _, t := range source.Spec.ForProvider.GlobalSecondaryIndexes {
-		parameter := &targetv1beta1.GlobalSecondaryIndexParameters{
-			Name:             t.IndexName,
-			NonKeyAttributes: t.Projection.NonKeyAttributes,
-			ProjectionType:   t.Projection.ProjectionType,
-		}
-		for _, a := range t.KeySchema {
-			if *a.KeyType == "HASH" {
-				parameter.HashKey = a.AttributeName
-			}
-			if *a.KeyType == "RANGE" {
-				parameter.RangeKey = a.AttributeName
+	if source.Spec.ForProvider.GlobalSecondaryIndexes != nil {
+		for _, t := range source.Spec.ForProvider.GlobalSecondaryIndexes {
+			if t.IndexName == nil {
```

Comment:
I am asking only to understand and ensure we are doing the right conversion. Is this check the correct one? Now, we are doing the assignment process if the IndexName is nil. Should we do this in this case, or should we do this if it is not nil?

Reply:
my bad @sergenyalcin, I already fixed it.